dosing_pump_controller/dosing_pump_controller.ino

```diff
@@ -78,77 +99,63 @@ void pump2CalibrationCompletion() { calibrate_pump2.calibrationComplete(); }
-  pump2_menu.addMenuItem(&calibrate_pump2, &pump2MenuReturnControl);
-  pump2_menu.addMenuItem(new ResetPumpSettingsMenuItem(pump2_controller), &pump2MenuReturnControl);
-
-  // TODO: Add items to the pump3_menu
-  pump3_menu.addMenuItem(new ReturnMenuItem(), &pump3MenuReturnControl);
-  pump3_menu.addMenuItem(new SetDoseMenuItem(pump3_controller), &pump3MenuReturnControl);
-  pump3_menu.addMenuItem(new EnablePumpMenuItem(pump3_controller), &pump3MenuReturnControl);
-  pump3_menu.addMenuItem(new PrimePumpMenuItem(pump3_controller), &pump3MenuReturnControl);
-  pump3_menu.addMenuItem(&calibrate_pump3, &pump3MenuReturnControl);
-  pump3_menu.addMenuItem(new ResetPumpSettingsMenuItem(pump3_controller), &pump3MenuReturnControl);
-
-  // TODO: Add items to the settings_menu
-  settings_menu.addMenuItem(new ReturnMenuItem(), &settingsMenuReturnControl);
-  settings_menu.addMenuItem(new BrightnessMenuItem(display_controller), &settingsMenuReturnControl);
-  settings_menu.addMenuItem(new ContrastMenuItem(display_controller), &settingsMenuReturnControl);
-
-  main_menu.present();
-  main_menu.sendAction(RotaryEncoder::Actions::select);
+  // Setup display and controllers.
+  board.display.setup();
+  board.display_controller.setup();
+  board.rotary_encoder.setup(&encoderAInput, &encoderBInput, &buttonInput, &actionDetected);
+  board.pump1.controller.setup(&pump1CalibrationCompletion);
+  board.pump2.controller.setup(&pump2CalibrationCompletion);
+  board.pump3.controller.setup(&pump3CalibrationCompletion);
+
+  // Configure the main menu and add menu items.
+  board.main_menu.configurePresentable(&board.display, nullptr);
+  board.main_menu.addMenuItem(new StartDosingMenuItem(board.pump1.controller, board.pump2.controller, board.pump3.controller), &mainMenuReturnControl);
+  board.main_menu.addMenuItem(&board.pump1.menu, &mainMenuReturnControl);
+  board.main_menu.addMenuItem(&board.pump2.menu, &mainMenuReturnControl);
+  board.main_menu.addMenuItem(&board.pump3.menu, &mainMenuReturnControl);
+  board.main_menu.addMenuItem(&board.settings_menu, &mainMenuReturnControl);
+
+  // Add items to the pump menus.
+  addPumpMenuItems(board.pump1.menu, calibrate_pump1, &pump1MenuReturnControl);
+  addPumpMenuItems(board.pump2.menu, calibrate_pump2, &pump2MenuReturnControl);
+  addPumpMenuItems(board.pump3.menu, calibrate_pump3, &pump3MenuReturnControl);
+
+  //Add items to the settings_menu
+  board.settings_menu.addMenuItem(new ReturnMenuItem(), &settingsMenuReturnControl);
+  board.settings_menu.addMenuItem(new BrightnessMenuItem(board.display_controller), &settingsMenuReturnControl);
+  board.settings_menu.addMenuItem(new ContrastMenuItem(board.display_controller), &settingsMenuReturnControl);
+
+  // Present the main menu and start dosing.
+  board.main_menu.present();
+  board.main_menu.sendAction(RotaryEncoder::Actions::select);
 }
 
 void loop() {
   unsigned long current_time = millis();
```

Comment:
Not introduced in this PR but if we want to use this long term we should look for an alternative to using millis() which will overflow after 49 days and might cause some timing issues.